src/Microsoft.TestPlatform.CommunicationUtilities/TcpClientExtensions.cs

```diff
@@ -40,17 +41,30 @@ internal static class TcpClientExtensions
                 socketException);
         }
 
+        // PERF: check if verbose is enabled once, and re-use for all calls in the tight loop below. The check for verbose is shows in perf traces
```

Comment:
This shown up as small overhead in VS monitoring, but the fact that it shown up says that the check is way more expensive than I would like for a message that would not be logged 99% of time.